File: 2019-ECP/workflow-examples/example10/compute.lua
#!/usr/bin/env lua

local f = assert (require 'flux' .new ())
local amount = tonumber (arg[1]) or 120

local function sleep (n)
    os.execute ("sleep " .. n)
end

if #arg ~= 1 then
   print ("Usage: compute.lua seconds")
   print ("    Compute for seconds")
   os.exit (1)
end

-- conduit is the KVS key that conduit module puts to publishing 
-- the rank in which it is loaded
cr = f:kvs_get ("conduit") or -1

if cr == -1 then
    print ("conduit key is not available")
    os.exit (1)
end

payload = '{"' .. os.time () .. '": "os.time"}'
print ("Sending " .. payload)

-- this data is ultimately flowed into the data store
local rc, err = f:send ("conduit.put", { data  = payload }, cr)
if rc < 0 then
    print ("error sending a conduit.put request")
    os.exit (1)
end

print ("Will compute for " .. amount .. " seconds")
sleep (amount) 

-- vi: ts=4 sw=4 expandtab

File: 2020-ECP/workflow-examples/example9/compute.py
#!/usr/bin/env python

import argparse
import time
import os
import re
import flux
import json
from flux import kvs
from flux.message import Message

parser = argparse.ArgumentParser(description="compute for seconds")
parser.add_argument(
    "integer",
    metavar="S",
    type=int,
    help="an integer for the number of seconds to compute",
)

args = parser.parse_args()

f = flux.Flux()
udata = "conduit"
kvs.put(f, "conduit", udata)
kvs.commit(f)

cr = kvs.get(f, "conduit")
print(cr)

os_time = int(time.time())
payload = {str(os_time): "os.time"}
new_payload = {"data": json.dumps(payload)}
print("Sending ", json.dumps(new_payload))

# this data is ultimately flowed into the data store
f.rpc("conduit.put", new_payload, 0)


time.sleep(args.integer)
